packages/ui-components/src/components/tooltip/tooltip.tsx

```diff
@@ -30,6 +34,51 @@ export class KvTooltip implements ITooltip {
 		return merge({}, { placement: this.position }, this.options);
 	};
 
+	private getTooltipElement = (): HTMLElement | null => {
+		return this.el.shadowRoot.querySelector('#tooltip') as HTMLElement | null;
+	};
+
+	private getContentElement = (): HTMLElement | null => {
+		return this.contentElement ?? (this.el.shadowRoot.querySelector('#content') as HTMLElement | null);
+	};
+
+	private showTooltip = () => {
+		if (!this.disabled) {
+			const tooltip = this.getTooltipElement();
+			tooltip.style.display = 'inline-block';
+			this.update();
+		}
+	};
+
+	private hideTooltip = () => {
+		const tooltip = this.getTooltipElement();
+		tooltip.style.display = '';
+	};
+
+	private update = () => {
+		const tooltip = this.getTooltipElement();
+		const child = this.getContentElement();
+
+		computePosition(child, tooltip, this.getOptions()).then(({ x, y }) => {
+			tooltip.style.left = `${x}px`;
+			tooltip.style.top = `${y}px`;
+		});
+	};
+
+	componentDidRender() {
+		const child = this.getContentElement();
+		const tooltip = this.getTooltipElement();
+
+		if (tooltip === null) {
+			return;
+		}
+
+		child.addEventListener('mouseenter', this.showTooltip);
```

Comment:
Just one last thing, you need to ensure that this event is properly removed and it's not duplicated.
In order to do that you need two things:
- On `disconnectCallback` remove the event listeners.
- Before assigning these listeners please remove the previous ones.

Also, these create private methods to help you:
- `private listenToEvents(child: HTMLElement)`
- `private unlistenToEvents(child: HTMLElement)`

Reply:
Thanks! Will do an update 😃